include/libpmemobj++/string_view.hpp

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+#ifndef LIBPMEMOBJ_CPP_STRING_VIEW
+#define LIBPMEMOBJ_CPP_STRING_VIEW
+
+#include <string>
+
+#if __cpp_lib_string_view
+#include <string_view>
+#endif
+
+namespace pmem
+{
+
+namespace obj
+{
+
+#if __cpp_lib_string_view
+using string_view = std::string_view;
+#else
+/*! \class string_view
+	\brief Our brief std::string_view implementation.
+
+	If C++17's std::string_view implementation is not available, this one is
+   used to avoid unnecessary string copying.
+*/
+class string_view {
```

Comment:
is this the same class as in pmemkv?
This should probably be a separate patch + a follow-up in pmemkv that removes its implementation.